`ClassCastException` in spoon

### Summary

`Autograder failed: de.firemage.autograder.core.LinterException: java.lang.ClassCastException: class spoon.support.reflect.code.CtForEachImpl cannot be cast to class spoon.reflect.code.CtAssignment (spoon.support.reflect.code.CtForEachImpl and spoon.reflect.code.CtAssignment are in unnamed module of loader 'app')`

### Lint Name

`RedundantUninitializedVariable`

### Reproducer

Does not appear to be dependent on a specific submission.

### Error output

<details><summary>Backtrace</summary>
  <p>

  ```
!ENTRY edu.kit.kastel.sdq.eclipse.grading.view 4 0 2023-10-07 15:10:24.428
!MESSAGE Autograder failed: de.firemage.autograder.core.LinterException: java.lang.ClassCastException: class spoon.support.reflect.code.CtForEachImpl cannot be cast to class spoon.reflect.code.CtAssignment (spoon.support.reflect.code.CtForEachImpl and spoon.reflect.code.CtAssignment are in unnamed module of loader 'app')
	at de.firemage.autograder.core.Linter.checkFile(Linter.java:243)
	at de.firemage.autograder.core.Linter.checkFile(Linter.java:153)
	at de.firemage.autograder.cmd.Application.execute(Application.java:138)
	at de.firemage.autograder.cmd.Application.call(Application.java:254)
	at de.firemage.autograder.cmd.Application.call(Application.java:48)
	at picocli.CommandLine.executeUserObject(CommandLine.java:2041)
	at picocli.CommandLine.access$1500(CommandLine.java:148)
	at picocli.CommandLine$RunLast.executeUserObjectOfLastSubcommandWithSameParent(CommandLine.java:2461)
	at picocli.CommandLine$RunLast.handle(CommandLine.java:2453)
	at picocli.CommandLine$RunLast.handle(CommandLine.java:2415)
	at picocli.CommandLine$AbstractParseResultHandler.execute(CommandLine.java:2273)
	at picocli.CommandLine$RunLast.execute(CommandLine.java:2417)
	at picocli.CommandLine.execute(CommandLine.java:2170)
	at de.firemage.autograder.cmd.Application.runApplication(Application.java:114)
	at de.firemage.autograder.cmd.Application.main(Application.java:106)
Caused by: java.lang.ClassCastException: class spoon.support.reflect.code.CtForEachImpl cannot be cast to class spoon.reflect.code.CtAssignment (spoon.support.reflect.code.CtForEachImpl and spoon.reflect.code.CtAssignment are in unnamed module of loader 'app')
	at spoon.reflect.visitor.chain.CtQueryImpl$1.apply(CtQueryImpl.java:171)
	at spoon.reflect.visitor.chain.CtQueryImpl$1.apply(CtQueryImpl.java:168)
	at spoon.reflect.visitor.chain.CtQueryImpl$FunctionWrapper._accept(CtQueryImpl.java:510)
	at spoon.reflect.visitor.chain.CtQueryImpl$AbstractStep.accept(CtQueryImpl.java:305)
	at spoon.reflect.visitor.filter.CtScannerFunction$Scanner.onElement(CtScannerFunction.java:78)
	at spoon.reflect.visitor.EarlyTerminatingScanner.doScan(EarlyTerminatingScanner.java:141)
	at spoon.reflect.visitor.EarlyTerminatingScanner.scan(EarlyTerminatingScanner.java:121)
	at spoon.reflect.visitor.filter.CtScannerFunction.apply(CtScannerFunction.java:57)
	at spoon.reflect.visitor.filter.CtScannerFunction.apply(CtScannerFunction.java:27)
	at spoon.reflect.visitor.chain.CtQueryImpl$LazyFunctionWrapper._accept(CtQueryImpl.java:490)
	at spoon.reflect.visitor.chain.CtQueryImpl$AbstractStep.accept(CtQueryImpl.java:305)
	at spoon.reflect.visitor.chain.CtQueryImpl.forEach(CtQueryImpl.java:97)
	at spoon.reflect.visitor.chain.CtQueryImpl.list(CtQueryImpl.java:110)
	at spoon.reflect.visitor.chain.CtQueryImpl.list(CtQueryImpl.java:104)
	at spoon.support.reflect.declaration.CtElementImpl.getElements(CtElementImpl.java:330)
	at de.firemage.autograder.core.check.complexity.RedundantUninitializedVariable$1.process(RedundantUninitializedVariable.java:49)
	at de.firemage.autograder.core.check.complexity.RedundantUninitializedVariable$1.process(RedundantUninitializedVariable.java:39)
	at spoon.support.visitor.ProcessingVisitor.scan(ProcessingVisitor.java:72)
	at spoon.reflect.visitor.CtScanner.scan(CtScanner.java:184)
	at spoon.reflect.visitor.CtScanner.visitCtForEach(CtScanner.java:494)
	at spoon.support.reflect.code.CtForEachImpl.accept(CtForEachImpl.java:30)
	at spoon.reflect.visitor.CtScanner.scan(CtScanner.java:194)
	at spoon.support.visitor.ProcessingVisitor.scan(ProcessingVisitor.java:68)
	at spoon.reflect.visitor.CtScanner.scan(CtScanner.java:184)
	at spoon.reflect.visitor.CtScanner.scan(CtScanner.java:149)
	at spoon.reflect.visitor.CtScanner.visitCtBlock(CtScanner.java:321)
	at spoon.support.reflect.code.CtBlockImpl.accept(CtBlockImpl.java:58)
	at spoon.reflect.visitor.CtScanner.scan(CtScanner.java:194)
	at spoon.support.visitor.ProcessingVisitor.scan(ProcessingVisitor.java:68)
	at spoon.reflect.visitor.CtScanner.scan(CtScanner.java:184)
	at spoon.reflect.visitor.CtScanner.visitCtMethod(CtScanner.java:589)
	at spoon.support.reflect.declaration.CtMethodImpl.accept(CtMethodImpl.java:59)
	at spoon.reflect.visitor.CtScanner.scan(CtScanner.java:194)
	at spoon.support.visitor.ProcessingVisitor.scan(ProcessingVisitor.java:68)
	at spoon.reflect.visitor.CtScanner.scan(CtScanner.java:184)
	at spoon.reflect.visitor.CtScanner.scan(CtScanner.java:149)
	at spoon.reflect.visitor.CtScanner.visitCtClass(CtScanner.java:357)
	at spoon.support.reflect.declaration.CtClassImpl.accept(CtClassImpl.java:63)
	at spoon.reflect.visitor.CtScanner.scan(CtScanner.java:194)
	at spoon.support.visitor.ProcessingVisitor.scan(ProcessingVisitor.java:68)
	at spoon.reflect.visitor.CtScanner.scan(CtScanner.java:184)
	at spoon.reflect.visitor.CtScanner.scan(CtScanner.java:149)
	at spoon.reflect.visitor.CtScanner.visitCtPackage(CtScanner.java:679)
	at spoon.support.reflect.declaration.CtPackageImpl.accept(CtPackageImpl.java:82)
	at spoon.reflect.visitor.CtScanner.scan(CtScanner.java:194)
	at spoon.support.visitor.ProcessingVisitor.scan(ProcessingVisitor.java:68)
	at spoon.reflect.visitor.CtScanner.scan(CtScanner.java:184)
	at spoon.reflect.visitor.CtScanner.scan(CtScanner.java:149)
	at spoon.reflect.visitor.CtScanner.visitCtPackage(CtScanner.java:678)
	at spoon.support.reflect.declaration.CtPackageImpl.accept(CtPackageImpl.java:82)
	at spoon.reflect.visitor.CtScanner.scan(CtScanner.java:194)
	at spoon.support.visitor.ProcessingVisitor.scan(ProcessingVisitor.java:68)
	at spoon.reflect.visitor.CtScanner.scan(CtScanner.java:184)
	at spoon.reflect.visitor.CtScanner.scan(CtScanner.java:149)
	at spoon.reflect.visitor.CtScanner.visitCtPackage(CtScanner.java:678)
	at spoon.support.reflect.declaration.CtPackageImpl.accept(CtPackageImpl.java:82)
	at spoon.reflect.visitor.CtScanner.scan(CtScanner.java:194)
	at spoon.support.visitor.ProcessingVisitor.scan(ProcessingVisitor.java:68)
	at spoon.reflect.visitor.CtScanner.scan(CtScanner.java:184)
	at spoon.reflect.visitor.CtScanner.scan(CtScanner.java:149)
	at spoon.reflect.visitor.CtScanner.visitCtPackage(CtScanner.java:678)
	at spoon.support.reflect.declaration.CtPackageImpl.accept(CtPackageImpl.java:82)
	at spoon.reflect.visitor.CtScanner.scan(CtScanner.java:194)
	at spoon.support.visitor.ProcessingVisitor.scan(ProcessingVisitor.java:68)
	at spoon.reflect.visitor.CtScanner.scan(CtScanner.java:184)
	at spoon.reflect.visitor.CtScanner.scan(CtScanner.java:149)
	at spoon.reflect.visitor.CtScanner.visitCtPackage(CtScanner.java:678)
	at spoon.support.reflect.declaration.CtPackageImpl.accept(CtPackageImpl.java:82)
	at spoon.reflect.visitor.CtScanner.scan(CtScanner.java:194)
	at spoon.support.visitor.ProcessingVisitor.scan(ProcessingVisitor.java:68)
	at spoon.reflect.visitor.CtScanner.scan(CtScanner.java:184)
	at spoon.reflect.visitor.CtScanner.visitCtModule(CtScanner.java:964)
	at spoon.reflect.factory.ModuleFactory$CtUnnamedModule.accept(ModuleFactory.java:96)
	at spoon.reflect.visitor.CtScanner.scan(CtScanner.java:194)
	at spoon.support.visitor.ProcessingVisitor.scan(ProcessingVisitor.java:68)
	at spoon.support.QueueProcessingManager.process(QueueProcessingManager.java:118)
	at spoon.reflect.CtModelImpl.processWith(CtModelImpl.java:126)
	at de.firemage.autograder.core.CodeModel.processWith(CodeModel.java:84)
	at de.firemage.autograder.core.integrated.StaticAnalysis.processWith(StaticAnalysis.java:36)
	at de.firemage.autograder.core.check.complexity.RedundantUninitializedVariable.check(RedundantUninitializedVariable.java:39)
	at de.firemage.autograder.core.integrated.IntegratedCheck.run(IntegratedCheck.java:34)
	at de.firemage.autograder.core.integrated.IntegratedAnalysis.lambda$lint$0(IntegratedAnalysis.java:107)
	at de.firemage.autograder.core.parallel.AnalysisThread.run(AnalysisThread.java:43)
	at java.base/java.lang.Thread.run(Thread.java:1583)
  ```

  </p>
</details>

--- a/autograder-core/pom.xml
+++ b/autograder-core/pom.xml
@@ -13,7 +13,7 @@
     <name>autograder-core</name>
 
     <properties>
-        <spoon.version>10.4.1</spoon.version>
+        <spoon.version>10.4.2</spoon.version>
         <pmd.version>7.0.0-rc3</pmd.version>
         <spotbugs.version>6.45.0</spotbugs.version>
 

